From 0fbdeaff336234574b3d05747b4c873d4abf95a2 Mon Sep 17 00:00:00 2001
From: Eduardo Habkost <ehabkost@redhat.com>
Date: Fri, 10 Apr 2026 12:43:52 -0300
Subject: [PATCH] STX: Add support statement to -help output

Add support statement to -help output, reporting direct qemu-kvm usage
as unsupported by StarlingX, and advising users to use libvirt instead.

File moved from softmmu/vl.c to system/vl.c in qemu 10.0.8.

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
Signed-off-by: Rafael Falcao <Rafael.VieiraFalcao@windriver.com>
[Rebased original changes to the qemu 7.2 version]
Signed-off-by: david.liu <david.liu@windriver.com>
[Rebased to qemu 10.0.8 for Trixie]
Signed-off-by: cbohlhal <Caique.BohlhalterdeSouza@windriver.com>
---
 system/vl.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/system/vl.c b/system/vl.c
index ec93988a03..fdc664ba16 100644
--- a/system/vl.c
+++ b/system/vl.c
@@ -870,9 +870,17 @@ static void version(void)
            QEMU_COPYRIGHT "\n");
 }
 
+static void print_stx_warning(void)
+{
+    printf("\nWARNING: Direct use of qemu-kvm from the command line is not supported by StarlingX.\n"
+           "WARNING: Use libvirt as the stable management interface.\n"
+           "WARNING: Some command line options listed here may not be available in future releases.\n\n");
+}
+
 static void help(int exitcode)
 {
     version();
+    print_stx_warning();
     printf("usage: %s [options] [disk_image]\n\n"
            "'disk_image' is a raw hard disk image for IDE hard disk 0\n\n",
             g_get_prgname());
@@ -898,6 +906,7 @@ static void help(int exitcode)
            "\n"
            QEMU_HELP_BOTTOM "\n");
 
+    print_stx_warning();
     exit(exitcode);
 }
 
-- 
2.34.1

